Sources/Stores/NetworkPersistableStore.swift

```diff
@@ -134,48 +144,55 @@ where Network.Remote == Data, Persistence.Remote == Data  {
 
         do {
             // Check if it's cancelled
-            guard cancelable.isCancelled == false else { return completion(nil, Error.cancelled) }
+            guard cancelable.isCancelled == false else {
+                completion(.failure(.cancelled))
+                return
+            }
 
             // parse the new value from the data
             let value = try parse(data: data, for: resource)
 
             // Check if it's cancelled
-            guard cancelable.isCancelled == false else { return completion(nil, Error.cancelled) }
+            guard cancelable.isCancelled == false else {
+                completion(.failure(.cancelled))
+                return
+            }
 
             // update persistence with new value
             if !fromCache {
                 self.persist(data, for: resource)
             }
 
-            completion(fromCache ? .persistence(value) : .network(value), nil)
+            completion(.success(fromCache ? .persistence(value) : .network(value)))
 
         } catch let error as Parse.Error {
-            completion(nil, Error.parse(error))
+            completion(.failure(.parse(error)))
         } catch {
-            completion(nil, Error.other(error))
+            completion(.failure(.other(error)))
         }
     }
 
     // MARK: Network Methods
 
-    private func getNetworkData<R>(_ resource: R, completion: @escaping (Data?, Error?) -> ())
+    private func getNetworkData<R>(_ resource: R, completion: @escaping (Data?, Error?) -> Void)
```

Comment:
`completion: @escaping (Result<Data, Error>) -> Void`?

Reply:
Well spotted! Fixing it.

Reply:
Done. ✨